### ISSUE TITLE
All translated sentences start with a same word

### ISSUE THREAD

**Issue by zhangdistephen** [jadore801120/attention-is-all-you-need-pytorch#44]
![image](https://user-images.githubusercontent.com/25263843/37661119-01b25cc2-2c8f-11e8-9cd3-90219f860a0a.png)

I wonder has anyone run the code and whether encounter the same problem? @ZiJianZhao

**Comment by juditacs** [comment]
Any update on this? We are experiencing the same issue for morphological reinflection.

~~~
k l l a m i g a z g a t á s b a n </s>
k a c s k a e l e d e l h e z </s>
k o m m u n i s t á v á </s>
k g y o n v á g n á k </s>
k o k o z t a t o k </s>
k e r e m t é s r e </s>
k a r n i s n á l </s>
k z é r i á é r t </s>
k a b o l t á t o k </s>
~~~

**Comment by JulianRMedina** [comment]
I forked this repo to my own and will be fixing things with the model in general. If you want to help with that process please see the repo here: https://github.com/JulianRMedina/attention-is-all-you-need-pytorch

**Comment by juditacs** [comment]
I will be paying attention your work.

I can't promise direct help right now but I would be happy to share some morphological data that is not a a toy dataset but still very easy and the model should be able to learn it quickly (the standard seq2seq converges in a few epochs and performs with over 90% accuracy).

**Comment by saitarslanboun** [comment]
I think their is similar problem for Multi30K dataset. For english-german translation, all the sentences with start with "Ein" or "Einige". And on training set, "Ein" or "Einige" does not appear overwhelmingly at the beginning of the sentences.

**Comment by kannadaraj** [comment]
Same Issue. for me too.. All words are starting with same word. What might be the possible reason...?

**Comment by saitarslanboun** [comment]
I solved the problem.

If you change
```
# wrap into a Variable
dec_partial_seq = Variable(dec_partial_seq, volatile=True)
```
with 

```
# wrap into a Variable
dec_partial_seq = Variable(dec_partial_seq, volatile=True)
     if i == 0:
          s = torch.zeros(dec_partial_seq.size()).cuda().fill_(2).long()
          dec_partial_seq = torch.cat((dec_partial_seq, s), dim=1)
```

in "transformer/Translator.py" file, the problem is solved.

Best,

**Comment by yueyongjiao** [comment]
> ![image](https://user-images.githubusercontent.com/25263843/37661119-01b25cc2-2c8f-11e8-9cd3-90219f860a0a.png)
> 
> I wonder has anyone run the code and whether encounter the same problem? @ZiJianZhao

**Comment by yueyongjiao** [comment]
Excuse me, The problems you have solved? and  how？

**Comment by saitarslanboun** [comment]
I wrote it down on up. If you go to the version on the date when I wrote, you can do the change what I have told, and you will solve the problem.

**Comment by yueyongjiao** [comment]
If the code can be used in the source text length is larger than the length of the target text in the task?

**Comment by yueyongjiao** [comment]
May this code can be used in the task where the source text length is larger than the length of the target text，such  as summarization generation ?

**Comment by saitarslanboun** [comment]
It seems to me it is not pretty straightforward thing to do. You have to make some modifications in encoder part of the code.

**Comment by nhh1501** [comment]
I try to create chatbot. But all translated sentencest always same result

i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>
i m not not not not not not . </s>

**Comment by rush86999** [comment]
i m having the same issue with same translation, any ideas would be helpful

**Comment by jadore801120** [comment]
Hi all, please take a look on the newest code.
I made the beam search more concise and clear. Hope it helps.
